src/bidiMapper/domains/script/channelProxy.ts

```diff
@@ -59,94 +51,95 @@ export class ChannelProxy {
       );
     }
 
-    this.#channel = channel;
-    this.#eventManager = eventManager;
-    this.#realm = realm;
-    this.#channelHandle = channelHandle;
-    this.#sendMessageHandle = sendMessageHandle;
+    this.#properties = channel;
   }
 
-  static async init(
-    channel: Script.ChannelProperties,
-    eventManager: IEventManager,
-    realm: Realm
-  ): Promise<ChannelProxy> {
-    const channelHandle = await ChannelProxy.#createHandle(realm);
+  /**
+   * Creates a channel proxy in the given realm, initialises listener and
+   * returns a handle to `sendMessage` delegate.
+   * */
+  async init(realm: Realm, eventManager: IEventManager): Promise<Handle> {
+    const channelHandle = await ChannelProxy.#createAndGetHandleInRealm(realm);
     const sendMessageHandle = await ChannelProxy.#createSendMessageHandle(
       realm,
       channelHandle
     );
 
-    const channelProxy = new ChannelProxy(
-      channelHandle,
-      sendMessageHandle,
-      channel,
-      eventManager,
-      realm
-    );
+    void this.#startListener(realm, channelHandle, eventManager);
+    return sendMessageHandle;
+  }
 
-    void channelProxy.initChannelListener();
-    return channelProxy;
+  /** Gets a ChannelProxy from window and returns its handle. */
+  async startListenerFromWindow(realm: Realm, eventManager: IEventManager) {
+    const channelHandle = await this.#getHandleFromWindow(realm);
+    void this.#startListener(realm, channelHandle, eventManager);
   }
 
   /**
-   * Returns a handle to the delegate sending message.
+   * Evaluation string which creates a ChannelProxy object on the client side.
    */
-  get sendMessageHandle(): CommonDataTypes.Handle {
-    return this.#sendMessageHandle;
+  static #createChannelProxyEvalStr() {
+    const functionStr = String(() => {
+      const queue: unknown[] = [];
+      let queueNonEmptyResolver: null | (() => void) = null;
+
+      return {
+        /**
+         * Gets a promise, which is resolved as soon as a message occurs
+         * in the queue.
+         */
+        async getMessage(): Promise<unknown> {
+          const onMessage: Promise<void> =
+            queue.length > 0
+              ? Promise.resolve()
+              : new Promise<void>((resolve) => {
+                  queueNonEmptyResolver = resolve;
+                });
+          await onMessage;
+          return queue.shift();
+        },
+
+        /**
+         * Adds a message to the queue.
+         * Resolves the pending promise if needed.
+         */
+        sendMessage(message: string) {
```

Comment:
Here the queue is always adding strings.

Change one of:

- make it a queue of string
- make message have unknown type

For consistency.

Reply:
Made it `unknown`